### PR TITLE
[DOC] Adding development docs to make sure it is clear how to do dev

### PR DESCRIPTION
**Pull request by adam2392**
Currently, it is unclear sometimes how to setup, or carry out development locally. This document seeks to address this.

Please let me know if this addresses any confusion you have had, or have.

Here is the file rendered for your ease-of-review: https://github.com/neurodata/scikit-tree/blob/e1aafe47cb266e6fa74cab5ef6093a2cd4d60942/DEVELOPING.md

## Before submitting

- [ ] I've read and followed all steps in the [Making a pull request](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [ ] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [ ] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

## After submitting

- [ ] All GitHub Actions jobs for my pull request have passed.

### CODE REVIEW

File: DEVELOPING.md [hunk 1]
```diff
@@ -1,14 +1,97 @@
...
+
 # Requirements
-* Python 3.8+
-* Poetry (`curl -sSL https://install.python-poetry.org | python - --version=1.2.2`)
 
-For the other requirements, inspect the ``pyproject.toml`` file. If you are updated the dependencies, please run `poetry update` to update the
+* Python 3.9+
+* numpy
+* scipy
+* scikit-learn>=1.3.1
```
Comment (sampan501) [inline DEVELOPING.md:25]: I think we need minimum versions for numpy and scipy

File: DEVELOPING.md [hunk 2]
```diff
@@ -1,14 +1,97 @@
...
+
+For the other requirements, inspect the ``pyproject.toml`` file.
+
+# Setting up your development environment
+
+We recommend using miniconda, as python virtual environments may not setup properly compilers necessary for our compiled code. 
+
+<!-- Setup a conda env -->
+
+    conda create -n sktree python=3.9
```
Comment (sampan501) [inline DEVELOPING.md]: Don't think we need to specify python version. We could maybe link to how to create conda envirnoments before the code

File: DEVELOPING.md [hunk 3]
```diff
@@ -1,14 +1,97 @@
...
+    pip install -r test_requirements.txt
+    pip install -r doc_requirements.txt
+
+# Building the project from source
+
+We leverage meson to build scikit-tree from source. We utilize a CLI tool, called [spin](https://github.com/scientific-python/spin), which wraps certain meson commands to make building easier.
+
+For example, the following command will build the project
+
+    spin build
```
Comment (sampan501) [inline DEVELOPING.md:68]: What about `spin build --clean` to build from scratch

File: DEVELOPING.md [hunk 4]
```diff
@@ -1,14 +1,97 @@
...
+
+For other commands, see
+
+    spin --help
+
+Note at this stage, you will be unable to run Python commands directly. For example, ``pytest ./sktree`` will not work.
+
+However, after installing and building the project from source using meson, you can leverage editable installs to make testing code changes much faster.
+
+    pip install --no-build-isolation --editable .
```
Comment (sampan501) [inline DEVELOPING.md:82]: We could alternatively use `meson` right? Is this the preferred way?

**Review by sampan501 — CHANGES_REQUESTED** [review]
Few minor comments, but otherwise we should be good to merge

**Review by PSSF23 — APPROVED** [review]
For some reason the commands 2 & 3 in the last section are not rendered as code, but the new doc looks good.

File: DEVELOPING.md (hunk 1 above)
Comment (adam2392) [reply DEVELOPING.md:25]: Done

File: DEVELOPING.md (hunk 2 above)
Comment (adam2392) [reply DEVELOPING.md]: Okay linked

File: DEVELOPING.md (hunk 3 above)
Comment (adam2392) [reply DEVELOPING.md:68]: Added a brief comment on using this flag

File: DEVELOPING.md (hunk 4 above)
Comment (adam2392) [reply DEVELOPING.md:82]: The editable install unfortunately requires you to use the pip method. However, I've had issues running it w/o first building. So the suggested sequence ends up being:

./spin build
pip install -e ...

Comment (sampan501) [reply DEVELOPING.md:82]: ok sounds good!

**Review by sampan501 — APPROVED** [review]

**Review by SUKI-O — APPROVED** [review]

**Pull request merged**